Utt_Christelle_Homework_Software_Engineering_1.py

```diff
@@ -0,0 +1,138 @@
+"""
+TASK 1
+Write a class to represent a Cash Register.
+You class should keep the state of price total and purchased items
+Below is a starter code:
+------------------------
+1. you can add new variables and function if you want to
+2. you will NEED to add accepted method parameters where required.
+For example, method add_item probably accepts some kind of an item?..
+3. You will need to write some examples of how your code works.
+"""
+#PS! Printing is only for logging purposes.
+
+class CashRegister:
+
+    def __init__(self):
+        self.total_items = {} # {'item': price}
+        self.total_price = 0
+        self.discount = 0
+
+    def add_item(self, item, price):
+        if type(price) != int and type(price) != float:
+            return
+        price -= price * self.discount
```

Comment:
⚠️ The `add_item` method here violates the SRP principle. It has more than one responsiblities: 
1. Add item to dictionary
2. Calculate the total price.

Ideally you should split this into two functions.
1. add_item, should just add item to dictionary
2. get_total, this is where you calculate all the total price from the items dictionary